docs/adr/adr-004-state-interaction.md

```diff
@@ -0,0 +1,121 @@
+# ADR #004: Celestia-Node State Interaction for March 2022 Testnet
+
+<hr style="border:3px solid gray"> </hr>
+
+## Authors
+
+@renaynay
+
+## Changelog
+
+* 2022-01-14: initial draft
+
+<hr style="border:2px solid gray"> </hr>
+
+## Context
+
+Currently, celestia-node lacks a way for users to submit a transaction to celestia-core as well as access its own state
+and public state of others.
+
+## Decision
+
+Both celestia **light** and **full** nodes will run `StateService`. 
+`StateService` will be responsible for managing the RPC endpoints provided by `StateAccessor`.
+
+`StateAccessor` will be an interface for interacting with celestia-core in order to retrieve account-related information
+as well as submit transactions.
+
+### `StateService`
+
+`StateService` will expose several higher-level RPC endpoints for users to access the methods provided by the 
+`StateAccessor`.
+
+```go
+type StateService struct {
+    accessor StateAccessor
+}
+``` 
+
+### `StateAccessor`
+
+`StateAccessor` is defined by the two methods listed below and may be expanded in the future to include more methods 
+related to accounts and transactions in later iterations.
+
+```go
+type StateAccessor interface {
+    func AccountBalance(context.Context, Account) (Balance, error)
```

Comment:
From a quick look at the lens impl it seems this returns `sdk.Coins` which could be several balances (e.g. IBC coins or if the chain has several coins/tokens). We should consider returning only the Celes balance here.

Reply:
Like I described here? https://github.com/celestiaorg/celestia-node/pull/369/files#diff-51474eb9a6d4dcb6b29018687c8bfc9f38bda0259cac4562475c3a9a98c1568bR85

Reply:
Makes sense, yes. But even there you are not returning a `Coin`. Not sure if we have a `Balance` type anywhere. Or is that an alias?

Reply:
OK, I'm simply assuming balance is just a type alias or an abstraction for Coin. Ok for me to resolve this thread.

Reply:
Yes it will be an alias / abstraction.